### PR TITLE
PT - PD4

### PR DESCRIPTION
**Pull request by pawel99k**

### CODE REVIEW

**Comment by pawel99k** [comment]
Tak, to prawda, nie robiłem normalizacji zmiennych, bo w moim odczuciu (może się mylę) nie zawsze przynosi ona korzystne efekty, a dowodem na to są eksperymenty tworzone w trakcie pierwszego projektu. Co do RMSE, no to prawda, że takie suche liczby są trudne to ocenienia czy to dużo czy mało, nie wiedziałem z tym to można porównać. Czy otrzymany błąd i odniesienie do odchylenia standardowego zmiennej daje dobrą możliwość  porównawczą?

**Comment by MichalTurski** [comment]
Co do normalizacji: taka była treść tej pracy domowej żeby sprawdzić i porównać wyniki. I to tego się tyczy moje "changes requested", reszta to takie uwagi z kategorii "na przyszłość".

Odnośnie porównania z odchyleniem standardowym: daje nam to po prostu bardzo sensowny punkt odniesienia. I jest w tej samej jednostce, co bardzo istotne.  
Jest nawet metryka, która się opiera na tym koncepcie:  
https://scikit-learn.org/stable/modules/model_evaluation.html#explained-variance-score

**Comment by pawel99k** [comment]
Już rozumiem. Cały czas odbierałem 'skalowanie' z treści zadania w sensie 'skalowalnosci', tak jak skalowalne mogą być aplikacje, czyli czułości SVM na rozmiar danych wejściowych (stąd też mój komentarz przy jednym z parametrów) i stąd to nieporozumienie. Wprawdzie nie pasowało mi to do artykułu, ale nie przyszło mi do głowy, że chodzi o normalizację/standaryzację, nie słyszałem tego słowa w tym kontekście. Dzięki za feedback

**Comment by MichalTurski** [comment]
Słabe wyniki normalizacji (w sensie sklearna) są do przewidzenia. Dokumentacja mówi o tym narzędziu:   
```Normalize samples individually to unit norm.```   
Nic dziwnego, że to nie działa jak każdy element zbioru potraktowaliśmy zupełnie inną transformacją. Ta funkcja służy do zupełnie innego celu. Wykorzystalibyśmy ją gdybyśmy na przykład do modelu chcieli podawać wektor prawdopodobieństw (wtedy normalizacja L1 daje nam sumowanie do 1).
Jeśli chodzi o złe (a przynajmniej nie-lepsze) wyniki standaryzacji (znowu: w nomenklaturze sklearn): też jestem zaskoczony. Mam taką hipotezę, że w `test_pipeline()` zwracasz rezultat wywołania metody `fit()` czymkolwiek on jest. Dokumentacja niestety milczy. Być może on nie zwraca tego, co byśmy chcieli.
